### ISSUE TITLE
childEvents do not get triggered on LayoutViews for child views added with attachView

### ISSUE THREAD

**Issue by bradlord** [marionettejs/backbone.marionette#2568]
I am attaching a child view to a `LayoutView` by calling `attachView` on one of the layout's regions and defining a `childEvents` attribute for the layout to handle events from the child view. I expected the `childEvents` to work the same whether I call `region.show(view)` or `region.attachView(view)`, but it turns out that `childEvents` handlers are never called when the view is added via `attachView`.

It looks like the `childEvents` handlers are not called because `attachView` does not set the `_parent` attribute on the view. Manually setting `_parent` on the view makes `childEvents` work as expected. Is this the intended behavior?

I am seeing this issue in Marionette 2.4.1.

Here is some example code to demonstrate the issue:

https://jsfiddle.net/72vxrufh/1/

```
var ButtonView = Marionette.ItemView.extend({
    triggers: {
        'click': 'clicked'
    }
});

var Layout = Marionette.LayoutView.extend({
    el: '#layout',
    
    template: false,
    
    regions: {
        buttonRegion: '#button-region'
    },
    
    childEvents: {
        clicked: function() {
            this.$el.append('<div>clicked</div>')
        }
    },
    
    onRender: function() {
        var buttonView = new ButtonView({el: $('#button')});
        // Uncommenting the following makes child events work as expected
        // buttonView._parent = this.buttonRegion;
        this.buttonRegion.attachView(buttonView);
    }
});

new Layout().render();
```

**Comment by jasonLaster** [comment]
Nice sleuthing. This is definitely an issue and I think you're on the right path with the parent reference. 

Feel free to make a PR and we can evaluate. Also a PR with the failing test helps

**Comment by jasonLaster** [comment]
AttachView is honestly really neglected. 

I say that because it doesn't come up often. Also I am not sure if it should have the attach events. 

Another thought, should show call attachView?

Last comment, I'd support for adding _parent support to attachView. Both adding the reference to the new view and removing the reference for the old view

**Comment by ianmstew** [comment]
@bradlord I suspect this could be addressed with two additions to `attachView`, like @jasonLaster suggested.  Care to put up your second PR in a week?  :smile:

```js
delete this.currentView._parent;
view._parent = this;
```

@jasonLaster After taking a look, I'm thinking that `show()` calling `attachView` might get tricky, although we should completely reconsider the `show` method for `v3`.

**Comment by cmisenas** [comment]
Is this the designed behavior though? Currently I'm seeing the following in the [docs](https://github.com/marionettejs/backbone.marionette/blob/40f99e6935c740335c3d64c15dcef73f11d5b452/api/region.yaml#L300-L302)

```
  attachView: |
    Associate a new view with the region by setting the region's value of `currentView`. This will **not call render, show, or fire any events**.
```

I also just might not be getting how `attachView` is used (and if so, please explain :) ).

**Comment by ahumphreys87** [comment]
@bradlord im wondering what your use case is for using `attachView` rather than `show`? `attachView` doesn't actually do any DOM manipulation. I'm actually going to propose this method is cut from the API. @samccone @jasonLaster any objections?

**Comment by bradlord** [comment]
@ahumphreys87 My use case is more or less exactly what's described in the [documentation for `attachView`](http://marionettejs.com/docs/v2.4.2/marionette.region.html#attach-existing-view): I have an existing DOM structure, and I am attaching views to some DOM elements to add behavior. If I use `show` instead of `attachView`, the existing DOM elements will be removed.

@cmisenas I'm not expecting `attachView` to call render, show, or fire any events when. I'm expecting that the events fired by child views to be relayed up to the containing `LayoutView`, just like they are when the views are added via `show`.

**Comment by ahumphreys87** [comment]
ah! @ianmstew this is what we missed! :stuck_out_tongue: ok @bradlord I think its reasonable to expect events to be attached - @jasonLaster did you have concerns?

**Comment by jasonLaster** [comment]
i'm good with `attachView` binding the events. @cmisenas interested in taking a stab at this?

**Comment by cmisenas** [comment]
sure @jasonLaster. I think I have a bettter understanding now.

**Comment by ianmstew** [comment]
Fixed by #2675.
